client_sdk/go/fpc/crypto/crypto.go

```diff
@@ -29,16 +30,65 @@ import "C"
 
 var logger = flogging.MustGetLogger("fpc-client-crypto")
 
-func encrypt(input []byte, encryptionKey []byte) ([]byte, error) {
-	return input, nil
+func keyGen() ([]byte, error) {
+	// generate random symmetric key of 16 bytes
+	// the length is required by the pdo crypto library
+	key := make([]byte, 16)
+	n, err := rand.Read(key)
+	if n != len(key) || err != nil {
+		return nil, err
+	}
+
+	return key, nil
 }
 
-func keyGen() ([]byte, error) {
-	return []byte("fake key"), nil
+func encrypt(input []byte, encryptionKey []byte) ([]byte, error) {
+	//This is an RSA encryption performed with the pdo crypto library
+	//Importantly, the library uses 2048bit RSA keys, so the input size has to be ~200bytes
+	//TODO-1: bump up the key length to 3072
+	//TODO-2: extend procedure for large input sizes
+
+	inputMessagePtr := C.CBytes(input)
+	defer C.free(inputMessagePtr)
+
+	encryptionKeyPtr := C.CBytes(encryptionKey)
+	defer C.free(encryptionKeyPtr)
+
+	//pdo crypto uses 2048bit (256bytes) RSA keys, the encrypted message size buffer is set accordingly
+	const encryptedMessageSize = 256
```

Comment:
See above regarding use of const from PDO C code. Given that we also have a const for max message size, maybe also a good idea to test here and give a good corresponding log-message (enclave logs might not be built-in, in which case error tracing will be a bit more of a challenge ...)

Reply:
same answer as above, and added test